lib/src/main/java/com/dynatrace/metric/util/DimensionList.java

```diff
@@ -140,15 +140,37 @@ String serialize() {
     boolean firstIteration = true;
 
     for (Dimension dimension : dimensions) {
-      if (!firstIteration) {
-        builder.append(",");
-      } else {
-        firstIteration = false;
-      }
+      // if the dimension is not valid, don't add it to the serialized line
+      if (isDimensionValid(dimension)) {
```

Comment:
Nit: Could consider inverting the if to reduce nesting

Reply:
I was thinking about it, but I think the nesting isn't that bad here, and the continue might actually be more confusing. Open to change it though.